src/animatedRe.rei

```diff
@@ -225,35 +215,102 @@ module ValueXY: {
         ~iterations: int=?,
         unit
       ) =>
-      CompositeAnimation.t;
+      Animation.t;
   };
 };
 
 type animatedEvent;
 
 let event: (array('a), 'b) => animatedEvent;
 
-let delay: float => CompositeAnimation.t;
+let delay: float => Animation.t;
+
+let sequence: array(Animation.t) => Animation.t;
+
+let parallel: (array(Animation.t), {. "stopTogether": bool}) => Animation.t;
+
+let stagger: (float, array(Animation.t)) => Animation.t;
+
+let loop: (~iterations: int=?, ~animation: Animation.t, unit) => Animation.t;
+
+let createAnimatedComponent: ReasonReact.reactClass => ReasonReact.reactClass;
+
+let timing:
+  (
+    ~value: value(regular),
+    ~toValue: [ | `raw(float) | `animated(value(regular))],
+    ~easing: Easing.t=?,
+    ~duration: float=?,
+    ~delay: float=?,
+    ~isInteraction: bool=?,
+    ~useNativeDriver: bool=?,
+    ~onComplete: Animation.endCallback=?,
+    ~iterations: int=?,
+    unit
+  ) =>
+  Animation.t;
+
+let spring:
+  (
+    ~value: value(regular),
+    ~toValue: [ | `raw(float) | `animated(value(regular))],
+    ~restDisplacementThreshold: float=?,
+    ~overshootClamping: bool=?,
+    ~restSpeedThreshold: float=?,
+    ~velocity: float=?,
+    ~bounciness: float=?,
+    ~speed: float=?,
+    ~tension: float=?,
+    ~friction: float=?,
+    ~stiffness: float=?,
+    ~mass: float=?,
+    ~damping: float=?,
+    ~isInteraction: bool=?,
+    ~useNativeDriver: bool=?,
+    ~onComplete: Animation.endCallback=?,
+    ~iterations: int=?,
+    unit
+  ) =>
+  Animation.t;
+
+let decay:
+  (
+    ~value: value(regular),
+    ~velocity: float,
+    ~deceleration: float=?,
+    ~isInteraction: bool=?,
+    ~useNativeDriver: bool=?,
+    ~onComplete: Animation.endCallback=?,
+    ~iterations: int=?,
+    unit
+  ) =>
+  Animation.t;
 
-let sequence: array(CompositeAnimation.t) => CompositeAnimation.t;
+let stop: Animation.t => unit;
 
-let parallel:
-  (array(CompositeAnimation.t), {. "stopTogether": bool}) => CompositeAnimation.t;
+let start: (Animation.t, ~callback: Animation.endCallback=?, unit) => unit;
 
-let stagger: (float, array(CompositeAnimation.t)) => CompositeAnimation.t;
+let reset: Animation.t => unit;
 
-let loop: (~iterations: int=?, ~animation: CompositeAnimation.t, unit) => CompositeAnimation.t;
+[@deprecated "Please use Easing module instead"]
+module Easing = Easing;
 
-let createAnimatedComponent : ReasonReact.reactClass => ReasonReact.reactClass;
+[@deprecated
+  "Please use Animated.start, Animated.stop or Animated.reset instead"
+]
+module CompositeAnimation = Animation;
 
+[@deprecated "Please use Animated.timing instead"]
 module Timing = Value.Timing;
 
 module TimingXY = ValueXY.Timing;
```

Comment:
Do we want the deprecation comments on `*XY` modules, too?

Reply:
Good call.

Reply:
In this particular case, I ended up not deprecating the `*XY` modules because `spring`, `timing` and `decay` (top-level methods on this module) are for `Value`, not for `ValueXY` (as per React Native interface itself). To be honest, the `XY` to my knowledge after chatting with Facebook team is considered being deprecated in the future, so I wouldn't worry too much about it.